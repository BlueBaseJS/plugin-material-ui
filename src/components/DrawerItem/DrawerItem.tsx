import { DrawerItemProps, IconProps } from '@bluebase/components';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import MUIListItem from '@material-ui/core/ListItem';
import React from 'react';
import { getComponent } from '@bluebase/core';

export const DrawerItem = (props: DrawerItemProps) => {

	const { active, disabled, icon, onPress, right, title } = props;
	const Icon = getComponent<IconProps>('Icon', 'Noop');

	return (
		<MUIListItem button={!!onPress} disabled={disabled} selected={active} onClick={onPress}>
			{icon ? <ListItemIcon><Icon {...icon} /></ListItemIcon> : null}
			{(title) ? <ListItemText inset={!!icon} primary={title} /> : null}
			{right}
		</MUIListItem>
	);
};
