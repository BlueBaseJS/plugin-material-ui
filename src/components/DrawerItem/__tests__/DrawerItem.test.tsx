import { DrawerItem } from '../DrawerItem';
import React from 'react';
import { Text } from 'react-native';
import { shallow } from 'enzyme';

describe('DrawerItem', () => {

	test('should forward title prop as primary', () => {
		const component = shallow(
			<DrawerItem title="Item 1" />
		);
		expect(component.childAt(0).props().primary).toEqual('Item 1');
		expect(component.childAt(0).props().inset).toEqual(false);
	});

	test('should forward left and right as children, set inset to true', () => {
		const component = shallow(
			<DrawerItem title="Item 1" icon={{ name: 'rocket' }} right={<Text>R</Text>} />
			);
		expect(component).toMatchSnapshot();
		expect(component.childAt(0).childAt(0).props().name).toBe('rocket');
		expect(component.childAt(1).props().inset).toEqual(true);
		expect(component.childAt(2).childAt(0).text()).toEqual('R');
	});

	test('should not render anything in main area if no title or description', () => {
		const component = shallow(
			<DrawerItem title={null as any} icon={{ name: 'rocket' }} right={<Text>R</Text>} />
		);
		// expect(component).toMatchSnapshot();
		expect(component.childAt(0).childAt(0).props().name).toBe('rocket');
		expect(component.childAt(1).childAt(0).text()).toEqual('R');
	});

});
