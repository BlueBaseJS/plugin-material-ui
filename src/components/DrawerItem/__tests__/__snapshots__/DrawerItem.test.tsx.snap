// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerItem should forward left and right as children, set inset to true 1`] = `
<WithStyles(ListItem)
  button={false}
>
  <WithStyles(ListItemIcon)>
    <Icon_Noop
      name="rocket"
    />
  </WithStyles(ListItemIcon)>
  <WithStyles(ListItemText)
    inset={true}
    primary="Item 1"
  />
  <Text>
    R
  </Text>
</WithStyles(ListItem)>
`;
